src/python/pants/backend/python/macros/caof_utils.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import Any, Dict, Optional, Tuple, Union
+
+from packaging.utils import canonicalize_name as canonicalize_project_name
```

Comment:
Why rename? I usually find that confusing, unless it's necessary due to naming conflicts. If the original name is unclear then perhaps it should be renamed at source?

Reply:
We have a convention of renaming it